main.c

```diff
@@ -131,11 +135,11 @@ void openStream(struct vStreamArgs* args){
 			}
 		}
 
-		vStream.bufferEnd = true;
 
 		// Free the packet we allocated
 		av_free_packet(&packet);
 	}
+	vStream.bufferEnd = true;
```

Comment:
This was a source of issues.
The read thread would exit because the buffer end was marked before the file was complete. Moving this made it so the buffer end is only flagged when the file is finished being read.